src/conda_build_config.rs

```diff
@@ -0,0 +1,223 @@
+
+use minijinja::{Environment, Value};
+
+use crate::{
+    selectors::SelectorConfig,
+    variant_config::{VariantConfig, VariantConfigError},
+};
+
+#[derive(Debug)]
+struct ParsedLine {
+    content: String,
+    condition: Option<String>,
+}
+
+fn parse_line(line: &str) -> ParsedLine {
+    let parts: Vec<&str> = line.splitn(2, '#').collect();
+    let content = parts[0].trim().to_string();
+    let condition = parts.get(1).and_then(|c| {
+        if c.trim().starts_with('[') && c.trim().ends_with(']') {
+            Some(c.trim()[1..c.trim().len() - 1].trim().to_string())
+        } else {
+            None
+        }
+    });
+
+    ParsedLine { content, condition }
+}
+
+fn evaluate_condition(
+    condition: &str,
+    env: &Environment,
+    context: &BTreeMap<String, Value>,
+) -> bool {
+    if condition.is_empty() {
+        return true;
+    }
+    println!("Evaluate condition: {}", condition);
+    let template_str = format!("{{% if {} %}}true{{% else %}}false{{% endif %}}", condition);
+    let template = env.template_from_str(&template_str).unwrap();
+
+    template.render(context).unwrap() == "true"
+}
+
+/// Load an old-school conda_build_config.yaml file, and apply the selector_config to it
+/// The parser supports only a small subset of (potential) conda_build_config.yaml features.
+/// Especially, only `os.environ.get(...)` is supported.
+pub fn load_conda_build_config(
+    path: &Path,
+    selector_config: &SelectorConfig,
+) -> Result<VariantConfig, VariantConfigError> {
+    // load the text, parse it and load as VariantConfig using serde_yaml
+
+    let mut input = fs_err::read_to_string(path)
+        .map_err(|e| VariantConfigError::IOError(path.to_path_buf(), e))?;
+
+    let context = selector_config.clone().into_context();
+
+    let mut env = Environment::new();
+
+    env.add_function(
+        "environ_get",
+        move |name: String, default: Option<String>| {
+            let value = std::env::var(name).unwrap_or_else(|_| default.unwrap_or_default());
+            Ok(Value::from(value))
+        },
+    );
+
+    // replace all `os.environ.get` calls with `environ_get`
+    input = input.replace("os.environ.get", "environ_get");
+
+    let mut lines = Vec::new();
+    for line in input.lines() {
+        let parsed = parse_line(line);
+        let mut line_content = if let Some(condition) = &parsed.condition {
+            if evaluate_condition(condition, &env, &context) {
+                parsed.content.to_string()
+            } else {
+                continue;
+            }
+        } else {
+            parsed.content.to_string()
+        };
+
+        let trimmed = line_content.trim();
+        if let Some(node) = trimmed.strip_prefix("- ") {
+            let s = node.trim();
+            // try to parse as a float or integer
+            if s.parse::<f64>().is_ok() || s.parse::<i64>().is_ok() {
+                line_content = line_content.replace(s, &format!("\"{}\"", s));
```

Comment:
Is the replace needed? Why not just: 

```suggestion
                line_content = format!("- \"{}\"", s);
```

Reply:
indentation might be messed up in that case.